static/src/javascripts/projects/common/modules/commercial/ticker.js

```diff
@@ -0,0 +1,78 @@
+// @flow
+let count = 0;
+let interval;
+let goal;
+let total;
+
+const percentageTotalAsNegative = () => {
+    let percentage = (total / goal) * 100 - 100;
+    if (percentage > 0) {
+        percentage = 0;
+    }
+    return percentage;
+};
+
+const animateBar = () => {
+    const progressBarElement = document.querySelector(
+        '.epic-ticker__filled-progress'
+    );
+
+    if (progressBarElement && progressBarElement instanceof HTMLElement) {
+        progressBarElement.style.transform = `translateX(${percentageTotalAsNegative()}%)`;
+    }
+};
+
+const increaseCounter = () => {
+    count += Math.floor(total / 100);
+    const counterElement = document.querySelector(
+        '.epic-ticker__so-far .epic-ticker__count'
+    );
+
+    if (counterElement && counterElement instanceof HTMLElement) {
+        counterElement.innerHTML = `$${count.toLocaleString()}`;
+        if (count >= total) {
+            clearInterval(interval);
+            counterElement.innerHTML = `$${total.toLocaleString()}`;
+        }
+    }
+};
+
+const animateCount = () => {
+    if (interval === undefined) {
+        interval = setInterval(increaseCounter, 30);
+    }
+};
+
+const populateText = () => {
+    const goalElement = document.querySelector(
+        '.epic-ticker__goal .epic-ticker__count'
+    );
+
+    if (goalElement && goalElement instanceof HTMLElement) {
+        goalElement.innerHTML = `$${goalElement.toLocaleString()}`;
+    }
+};
+
+const getData = () => {
```

Comment:
This function could be named better, as it not only gets the data but also populates the text. Maybe it would be cleaner/clearer to the data as a promise then call `populateText` once they promise is resolved?

Reply:
yeah you're absolutely right. I only had a few hours on Friday to do this so I basically ES6-ified some old code of Sam's and it could do with some cleanup. to your other point about the `goal` being zero, yes - it's also not great that these functions depend on module-scope variables which have to be set in a certain order otherwise everything will blow up.

It works for now so I'm going to get it out as is and do some cleanup on it

Reply:
The other thing I should really do is use `requestAnimationFrame` - or maybe even CSS transitions - for smoother animation as this `setInterval` technique is not great performance-wise

Reply:
Actually ignore the last comment, I'm talking rubbish - it does use CSS transitions as it is. It's just a `setTimeout` before starting the transition

One nice thing I could do is not start the animation til the epic is in view.